src/core/core.tooltip.js

```diff
@@ -39,19 +39,13 @@ defaults._set('global', {
 		callbacks: {
 			// Args are: (tooltipItems, data)
 			beforeTitle: helpers.noop,
-			title: function(tooltipItems, data) {
-				// Pick first xLabel for now
+			title: function(tooltipItems) {
 				var title = '';
-				var labels = data.labels;
-				var labelCount = labels ? labels.length : 0;
 
 				if (tooltipItems.length > 0) {
 					var item = tooltipItems[0];
-
-					if (item.xLabel) {
-						title = item.xLabel;
-					} else if (labelCount > 0 && item.index < labelCount) {
```

Comment:
Is it safe to remove the part below? I don't really understand what these `xLabel` and `yLabel` options are doing. Perhaps you'd be able to improve the docs to better explain while you're at it

```
else if (labelCount > 0 && item.index < labelCount) {	
		title = labels[item.index];	
}
``

Reply:
item.index could be label index or data index, and therefore could be out of bounds.
`xLabel` and `yLabel` are same as `value` and `label`, just in arbitrary order (usually xLabel=label and yLabel=value, except for horizontalBar and the like).